src/app/WriteHandler.cpp

```diff
@@ -634,6 +645,52 @@ Status WriteHandler::ProcessWriteRequest(System::PacketBufferHandle && aPayload,
     return status;
 }
 
+CHIP_ERROR WriteHandler::VerifyAttributeDataIBs(const TLV::TLVReader & aAttributeDataIBsReader, bool aIsGroupAction)
```

Comment:
Why are we doing this in a separate validation pre-pass instead of just doing it while processing the IBs?

Reply:
This comes from the requirement from the spec:

Spec 8.7.3.2: If the attribute path is invalid, the processing should be terminated and a StatusIB should be generated.

Added a comment in the caller.

Reply:
> Spec 8.7.3.2: If the attribute path is invalid, the processing should be terminated and a StatusIB should be generated.

You're talking about this part, I assume:

> Else if the path does not conform to Valid Write Attribute Paths then:
> a Status Response with the INVALID_ACTION Status Code SHALL be generated as defined in Status Response Action

etc?  That is _specifically_ about wildcard paths that are wildcarding the wrong things.  I agree that per spec that needs to be a pre-check.  But that check is not what this PR is implementing...

Reply:
And actually, that spec requirement is a huge problem, because it's not really compatible with chunking the way chunking is implemented....  Going to raise a spec issue.

Reply:
Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5360

Reply:
@bzbarsky-apple @erjiaqing I think we may consider to enable CheckSchemaValidity for all platform, which can do pre-check for all messages, and enable prettyprint log for linux and darwin only in CheckSchemaValidity , suppress prettyprint log for the other platform for CheckSchemaValidity.

Reply:
I don't think we want all the extra code from enabling CheckSchemaValidity.  And again, per spec, this is not supposed to be a pre-check.

Reply:
Agreed. 

> But that check is not what this PR is implementing...

I think the spec is not super clear about handling mal-formed elements in the path itself.

Reply:
Then we should remove this pre check, and update the spec description. Thanks

Reply:
Updated the PR, our current behavior is already matching the discussions in 5360 and 5362.

So, the PR becomes "add a special check for invalid paths"